### PR TITLE
Correct Tx Integer Parsing for values > 32bit

### PR DESCRIPTION
**Pull request by ejfitzgerald**
- Fix issue with encoding and decoding values > 32bit
- Extract out integer encoding logic into own header file (in order to write tests)
- Add a test suite to specifically test the integer encoding scheme

### CODE REVIEW

File: libs/ledger/include/ledger/chain/transaction_encoding.hpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    if (is_signed && (abs_value <= 0x1F))
+    {
+      encoded.Resize(1);
+      encoded[0] = 0xE0 | static_cast<uint8_t>(abs_value);
+    }
+    else
+    {
+      // determine the corresponding number of bytes required to encode this value
+      uint64_t const bits_to_encode  = 64ull - platform::CountLeadingZeroes64(abs_value);
```
Comment (ejfitzgerald) [inline libs/ledger/include/ledger/chain/transaction_encoding.hpp:66]: This is basically the area of the code which had the logical flaw

File: libs/ledger/include/ledger/chain/transaction_encoding.hpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+      uint64_t partial_value{0};
+      for (std::size_t index = (1u << log2_value_length) - 1u;; --index)
+      {
+        // read the next byte
+        uint8_t encoded_byte{0};
+        buffer.ReadBytes(&encoded_byte, 1);
+
+        // build up the partial value
+        partial_value |= static_cast<uint64_t>(encoded_byte) << (index * 8u);
```
Comment (ejfitzgerald) [inline libs/ledger/include/ledger/chain/transaction_encoding.hpp:183]: This also had a logical flaw here

**Review by wkaluza — APPROVED** [review]

File: libs/ledger/include/ledger/chain/transaction_encoding.hpp [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+{
+  using fetch::byte_array::ByteArray;
+
+  bool const is_signed = meta::IsSignedInteger<T> && (value < 0);
+
+  ByteArray encoded;
+  if (!is_signed && (value <= T{0x7f}))
+  {
+    encoded.Resize(1);
+    encoded[0] = static_cast<uint8_t>(value & 0x7f);
```
Comment (pbukva) [inline libs/ledger/include/ledger/chain/transaction_encoding.hpp:52]: Nitpicking - the ` & 0x7f` masking should not be necessary (based on `if(...)` context)

File: libs/ledger/include/ledger/chain/transaction_encoding.hpp [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  }
+  else
+  {
+    uint64_t const abs_value = ToU64(value);
+
+    if (is_signed && (abs_value <= 0x1F))
```
Comment (pbukva) [inline libs/ledger/include/ledger/chain/transaction_encoding.hpp:58]: The `is_signed` can be dropped based on context (`else {...}`)

**Pull request closed without merging**